--- decompiled/screens/charSelect/CharacterOption.java
+++ decompiled/screens/charSelect/CharacterOption.java
@@ -34,6 +34,9 @@
 import com.megacrit.cardcrawl.unlock.UnlockTracker;
 import java.util.ArrayList;
 
+// ModLoader imports
+import modloader.ModLoader;
+
 public class CharacterOption {
     private static final UIStrings uiStrings = CardCrawlGame.languagePack.getUIString("CharacterOption");
     public static final String[] TEXT = CharacterOption.uiStrings.TEXT;
@@ -82,6 +85,10 @@
                 break;
             }
         }
+        
+        // ModLoader hook
+        if (CardCrawlGame.modLoaderActive) ModLoader.charOptionHook(charInfo);
+        
         this.hp = this.charInfo.hp;
         this.gold = this.charInfo.gold;
         this.flavorText = this.charInfo.flavorText;
